plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java

```diff
@@ -48,7 +48,6 @@ public static NessieIcebergClient createNessieIcebergClient(IcebergNessieCatalog
         return new NessieIcebergClient(
                 HttpClientBuilder.builder()
                         .withUri(icebergNessieCatalogConfig.getServerUri())
-                        .withEnableApiCompatibilityCheck(false)
```

Comment:
Why did you remove this line? The method was renamed to withApiCompatibilityCheck, not removed.

Reply:
This was added in the pervious Iceberg version bump as a temporary workaround for a bug in older Nessie version. It got fixed in the newer version. Hence, removed it. 

more info: https://github.com/trinodb/trino/pull/17715#discussion_r1213150069

Reply:
Thanks for explaining the context. It would be nice to leave such information in the commit body for future readers.